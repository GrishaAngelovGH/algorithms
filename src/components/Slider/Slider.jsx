import Slider from 'rc-slider'
import 'rc-slider/assets/index.css'

const SliderWrapper = ({ max, value, disabled, onChange }) => (
	<Slider
		min={0}
		max={max}
		value={value}
		disabled={disabled}
		onChange={onChange}
		railStyle={{
			height: 2
		}}
		handleStyle={{
			height: 28,
			width: 28,
			marginLeft: -14,
			marginTop: -14,
			backgroundColor: 'green',
			border: 0,
			opacity: 1
		}}
		trackStyle={{
			background: 'green'
		}}
	/>
)

export default SliderWrapper