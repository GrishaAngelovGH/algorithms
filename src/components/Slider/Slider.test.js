import { render } from '@testing-library/react'

import Slider from './Slider'

test('should render component', () => {
    const container = render(<Slider />)

    expect(container).toMatchSnapshot()
})