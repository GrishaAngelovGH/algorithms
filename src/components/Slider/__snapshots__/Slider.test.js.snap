// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rc-slider rc-slider-horizontal"
      >
        <div
          class="rc-slider-rail"
          style="height: 2px;"
        />
        <div
          class="rc-slider-track"
          style="left: 0%; width: 0%; background: green;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="left: 0%; transform: translateX(-50%); height: 28px; width: 28px; margin-left: -14px; margin-top: -14px; background-color: green; border: 0px; opacity: 1;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rc-slider rc-slider-horizontal"
    >
      <div
        class="rc-slider-rail"
        style="height: 2px;"
      />
      <div
        class="rc-slider-track"
        style="left: 0%; width: 0%; background: green;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle"
        role="slider"
        style="left: 0%; transform: translateX(-50%); height: 28px; width: 28px; margin-left: -14px; margin-top: -14px; background-color: green; border: 0px; opacity: 1;"
        tabindex="0"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
